discord_bot/bot.py

```diff
@@ -0,0 +1,136 @@
+"""
+This module defines a Discord bot with commands and utilities for sending messages,
+managing invites, and setting permissions in Discord channels.
+
+Classes:
+- DiscordBot: A custom Discord bot that extends `commands.Bot` to include additional functionality 
+such as sending messages, creating invite links, and managing channel permissions.
+
+Usage:
+Instantiate `DiscordBot` with a configuration and call `start_bot()` to run the bot.
+"""
+
+
+from typing import Any, Dict, Optional
+
+import discord
+from discord.ext import commands
+
+from .config import load_config
+
+
+class DiscordBot(commands.Bot):
+    def __init__(self, config: Dict[str, Any]) -> None:
+        self.config: Dict[str, Any] = config
+
+        # Define intents
+        intents = discord.Intents.default()
+        intents.messages = True
+        intents.guilds = True
+        intents.members = True
+
+        # Initialize the bot with the specified command prefix and intents
+        super().__init__(command_prefix="!", intents=intents)
+
+    async def on_ready(self) -> None:
+        """
+        Called when the bot is connected and ready. Logs connection details.
+        """
+
+        print(f"Bot connected as {self.user}")
+        for guild in self.guilds:
+            print(f"Connected to guild: {guild.name}")
+
+    async def send_message_to_channel(self, channel_name: str, message: str) -> None:
+        """
+        Sends a message to the specified discord channel by name.
+        """
+        await self.wait_until_ready()
+
+        channel = discord.utils.get(self.get_all_channels(), name=channel_name)
+        if channel:
+            await channel.send(message)
+        else:
+            print(f"Channel named '{channel_name}' not found")
```

Comment:
```suggestion
        if channel is not None:
            logger.warning("Channel named '%s' not found", channel_name)
            return
        await channel.send(message)
```
With this style you won't one day end up with
```python
if a:
    if b:
        if c:
            if d:.
                actual code but
                completely unreada-
                ble because of the exc-
                essive identation
            else:
                handle error d
        else:
            handle error c
    else:
        handle error b
else:
    handle error a, but if the code block is long enough, you can't really see error a condition anymore because it's more than a screen away and then you'd need to visually align the horizontal "else" with the "if" a screen or two above.
```
Early exit is better.
```python
if something is not right:
    logger.warning('reason')
    return False
if something else is not right:
    logger.warning('reason')
    return False
do the work
```

BTW use proper logger instead of prints.